pyteal/ast/frame.py

```diff
@@ -11,8 +14,101 @@
     from pyteal.compiler import CompileOptions
 
 
+@dataclass
+class LocalTypeSegment:
```

Comment:
yes, that are all good points to me:
- `__teal__()` method seems like a `Expr`: After some thoughts during the weekend, I believe we can attempt to make both `LocalTypeSegment` and `ProtoStackLayout` inherit from `Seq`, for they serve a similar purpose without revealing `type_of`. I hope this may help in sourcemapping!
- `dataclass` is just out of laziness for me unwilling to declare `__init__` function. For the simplicity and increasing complexity, I opt to change the class declaration to vanilla flavor of python class decl.

Reply:
I just took out the dependencies over `dataclasses` from `LocalTypeSegment` and `ProtoStackLayout`, and I hope (yet unsure how) it would help in sourcemapping.

Reply:
after a quick thought, I am taking out the dependencies to `Seq`, and let them directly inherit from `Expr`, just to decrease the complication.